### PR TITLE
Move Deduplicate P2P & Collective

### PR DESCRIPTION
**Pull request by AndyUB**
## What is done?

The tests for p2p and collective deduplication, custom comm deduplication, and custom comm init & teardown no longer require GPU.

Mock both init and teardown of NCCL groups.

## What is experimental?

Not sure if some channel creation depends on GPU.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/dag/tests/experimental/test_collective_dag.py [hunk 1]
```diff
@@ -45,6 +47,60 @@ def compute_with_tuple_args(self, args, i: int):
...
+        recv_buf: "torch.Tensor",
+        op: ReduceOp = ReduceOp.SUM,
+    ) -> None:
+        raise NotImplementedError
+
+    def destroy(self) -> None:
+        pass
+
+
+def mock_do_init_nccl_group(self, rank: int, nccl_group_id: str, custom_nccl_group: GPUCommunicator):
```
Comment (dengwxn) [inline python/ray/dag/tests/experimental/test_collective_dag.py]: Can this be moved inside the `MockNcclGroupSet` or its func?

Comment (dengwxn) [inline python/ray/dag/tests/experimental/test_collective_dag.py]: Can you tell me what's the difference between this and the actual `do_init_nccl_group`? In the actual one, `custom_nccl_group` is optional, why it isn't here?

File: python/ray/dag/tests/experimental/test_collective_dag.py [hunk 2]
```diff
@@ -63,6 +119,20 @@ def __call__(
             frozenset(actors),
             custom_nccl_group,
         )
+        if custom_nccl_group is None:
+            custom_nccl_group = AbstractNcclGroup(actors)
```
Comment (dengwxn) [inline python/ray/dag/tests/experimental/test_collective_dag.py]: Can this be None?

File: python/ray/dag/tests/experimental/test_collective_dag.py (hunk 1 above)
Comment (AndyUB) [reply python/ray/dag/tests/experimental/test_collective_dag.py]: Will change to optional.

File: python/ray/dag/tests/experimental/test_collective_dag.py (hunk 2 above)
Comment (AndyUB) [reply python/ray/dag/tests/experimental/test_collective_dag.py]: Yes. Will change.

File: python/ray/dag/tests/experimental/test_collective_dag.py (hunk 1 above)
Comment (AndyUB) [reply python/ray/dag/tests/experimental/test_collective_dag.py]: Yes.

Comment (AndyUB) [reply python/ray/dag/tests/experimental/test_collective_dag.py]: Reverted.

**Pull request merged**